CONTRIBUTING.md

```diff
@@ -56,34 +56,114 @@ requests - there's only so much [automated testing][ar:travis] can do.
 have access to, or try configurations not thought of before.
 
-## Patches
-
-Of course, we also accept patches. If you want to submit a patch, the
-guidelines are very, very simple:
-
- 1. Open an issue first, if there is none open for the particular
-    issue for the topic already in the
-    [issue tracker][ar:issue-tracker]. That way, other contributors
-    and developers can comment the issue and the patch.
- 2. If you submit a pull request that fixes an existing issue, mention
-    the issue somewhere in the pull request, so we can close the
-    original issue as well.
- 3. We are using a coding style very similar to
-    [GNU Coding Standards](https://www.gnu.org/prep/standards/standards.html#Writing-C)
-    for syslog-ng. Please try to follow the existing conventions.
- 4. Start your commit message with the following format:
-    `block-name: short description of the applied changes`.
-
-    Always add a `Signed-off-by` tag to the end of **every** commit
-    message you submit.
- 5. Always create a separate branch for the pull request, forked off
-    from the appropriate syslog-ng branch.
- 6. If your patch should be applied to multiple branches, submit
-    against the latest one only, and mention which other branches are
-    affected. There is no need to submit pull requests for each
-    branch.
- 7. If possible, write tests! We love tests.
- 8. A well-documented pull request is much easier to review and merge.
+## Pull requests
+
+If you plan to open a pull request, please follow the guideline below.
+
+### Branch
+
+You need to fork syslog-ng in github, and create a working branch from master in your fork.
+
+After writing the code, just before opening the pull request,
+make sure your working branch is rebased against master.
+
+### PR description
+In the description, please explain (if applicable):
+- The problem your pull request intends to solve.
+- A general overview about the implementation, or any information that can help
+reviewers to understand your code.
+- Please explain how one can try out your code: configuration snippets,
+or deployment information.
+- You can mention if you were considering alternative solutions or explain
+problems you ran into.
+- If you submit a pull request that fixes an existing issue, please mention
+the issue somewhere in the pull request, so we can close the
+original issue as well.
+
+
+The documentation is created from the description. Please provide
+a description that can be a good input for the admin guide as well.
+
+### Commits
+#### Commit messages
+
+```
+The commit messages be formatted according to this:
+
+module: short description
+
+Long description, that may be
+formatted in markdown.
+
+Signed-off-by: your name <youremail@address.com>
+```
+
+This format is checked by the CI.
+
+If you do not want to share your email address due to privacy reasons,
+you can use `yourgithubusername@users.noreply.github.com`,
```

Comment:
GitHub no longer generates these addresses, they use user IDs instead, for example, `1234567+username@users.noreply.github.com`.

But the gist is clear, so it's just a note.

Reply:
I reworded a bit.

Reply:
Thanks.